src/input/type_record.cc

```diff
@@ -124,9 +120,21 @@ void Record::make_derive_from(AbstractRecord &parent) {
     if (data_->derived_) return;
```

Comment:
Make a note , that this could by simplified if we manage to have TYPE key in every record. Currently, it is not possible due to new autoconversion rules for record transpose.